src/uu/cut/src/cut.rs

```diff
@@ -359,12 +359,22 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
 
     let complement = matches.get_flag(options::COMPLEMENT);
 
+    let mode_args_count = [
+        matches.indices_of(options::BYTES),
+        matches.indices_of(options::CHARACTERS),
+        matches.indices_of(options::FIELDS),
+    ]
+    .into_iter()
+    .filter_map(|mode| mode.map(|indices| indices.len()))
+    .sum();
```

Comment:
```suggestion
    let mode_args_count = [
        matches.indices_of(options::BYTES),
        matches.indices_of(options::CHARACTERS),
        matches.indices_of(options::FIELDS),
    ]
    .into_iter()
    .filter_map(|mode| mode.map(|indices| indices.len()))
    .sum();
```
I tried to find a way of expressing this without the nested map. This is what I landed on. Still the still in functionality though. I think this is a pretty good way of doing it!

Reply:
I was also wondering for more clean way to do it to remove nested map. It looks like the suggested change is identical to the original one, could you edit it to what you landed on?

Reply:
I don't know what @tertsdiepraam had in mind, here's an idea of how to get rid of the nested map:
```rust
let mode_args_count = [
    matches.indices_of(options::BYTES),
    matches.indices_of(options::CHARACTERS),
    matches.indices_of(options::FIELDS),
]
.into_iter()
.map(|i| i.unwrap_or_default().count())
.sum();
```

Reply:
Thanks @cakebaker! It looks like a nice way to transform `Option::None` to an empty iterator! Also, using `count()` instead of `len()` seems more correct way to do it, since it looks like the intended use case of `len()` is to get the remaining length of the iterator.
Just updated the code!

Reply:
I must've made a mistake while copying code from my editor. I'll take another looks in a bit.

Reply:
```suggestion
    let mode_args_count = [
        matches.indices_of(options::BYTES),
        matches.indices_of(options::CHARACTERS),
        matches.indices_of(options::FIELDS),
    ]
    .into_iter()
    .flatten()
    .map(|indices| indices.len()))
    .sum();
```
This is what I had in mind I think.